cdap-ui/app/cdap/components/AbstractWidget/FunctionDropdownArgumentsWidget/__tests__/parser.tests.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { parse, serialize } from 'components/AbstractWidget/FunctionDropdownArgumentsWidget/parser';
+
+describe('FunctionDropdownArgumentsRow', () => {
+  describe('parser', () => {
+    describe('parse', () => {
+      it('should parse a string with no arguments', () => {
+        expect(parse('myAlias:myFunc(field1,true)')).toEqual({
+          alias: 'myAlias',
+          func: 'myFunc',
+          field: 'field1',
+          ignoreNulls: true,
+          arguments: '',
+        });
+      });
+
+      it('should parse a string with ignoreNulls false', () => {
+        expect(parse('myAlias:myFunc(field1,false)')).toEqual({
+          alias: 'myAlias',
+          func: 'myFunc',
+          field: 'field1',
+          ignoreNulls: false,
+          arguments: '',
+        });
+      });
+
+      it('should parse a string with one argument', () => {
+        expect(parse(`myAlias:myFunc(field1,${encodeURIComponent('a')},true)`)).toEqual({
+          alias: 'myAlias',
+          func: 'myFunc',
+          field: 'field1',
+          ignoreNulls: true,
+          arguments: 'a',
+        });
+      });
+
+      it('should parse a string with multiple arguments', () => {
+        expect(parse(`myAlias:myFunc(field1,${encodeURIComponent('a,b,c')},true)`)).toEqual({
+          alias: 'myAlias',
+          func: 'myFunc',
+          field: 'field1',
+          ignoreNulls: true,
+          arguments: 'a,b,c',
+        });
+      });
+
+      it('should parse an argument with a colon', () => {
```

Comment:
Is it possible for an argument to include parentheses?

Reply:
Let me confirm with Bhooshan. For now, I'm assuming "no", since `encodeURLComponent` doesn't encode them.